arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -141,109 +143,136 @@ def start_pcp(
                 "The pmlogger output file was not found; Unable to process results."
             )
 
-        pcp2_flags = [
-            "-a",
-            "pmlogger-out",
-            "-t",
-            str(params.pmlogger_interval),
-            "-f",
-            # pmrep doesn't accept %z for the timezone,
-            # so we'll get it explicitly via datetime
-            f"%FT%T.%f{str(datetime.now().astimezone())[-6:]}",
-            "-c",
-            pmrep_conf_path,
-            "-P",
-            "6",
-        ]
-
-        # Initialized pcp2json command
-        pcp2json_cmd = [
-            "/usr/bin/pcp2json",
-            "-E",
-        ]
-
-        pcp2json_cmd.extend(pcp2_flags)
+        post_process_params = {
+            "pmlogger_metrics": params.pmlogger_metrics,
+            "pmlogger_interval": params.pmlogger_interval,
+            "pmrep_conf_path": params.pmrep_conf_path,
+            "generate_csv": params.generate_csv,
+            "flatten": params.flatten,
+            "archive_path": ".",
+        }
+
+        return post_process(
+            params=post_process_params_schema.unserialize(post_process_params),
+            run_id="post-process",
+        )
 
-        # The list of metrics to collect is appended to the pcp2json command
-        pcp2json_cmd.extend(metrics)
 
-        print(f"Reporting metrics for: {params.pmlogger_metrics}")
+@plugin.step(
+    id="post-process",
+    name="Post-Process PCP Archive",
+    description="Processes an existing PCP archive into a machine-readable format",
+    outputs={"success": PerfOutput, "error": Error},
+)
+def post_process(
+    params: PostProcessParams,
+) -> typing.Tuple[str, typing.Union[PerfOutput, Error]]:
+
+    # Parse metrics from input
+    metrics = params.pmlogger_metrics.split()
+
+    pcp2_flags = [
+        "-a",
+        params.archive_path,
+        "-t",
+        str(params.pmlogger_interval),
+        "-f",
+        # pmrep doesn't accept %z for the timezone,
+        # so we'll get it explicitly via datetime
+        f"%FT%T.%f{str(datetime.now().astimezone())[-6:]}",
+        "-c",
+        params.pmrep_conf_path,
+        "-P",
+        "6",
+    ]
+
+    # Initialized pcp2json command
+    pcp2json_cmd = [
+        "/usr/bin/pcp2json",
+        "-E",
+    ]
+
+    pcp2json_cmd.extend(pcp2_flags)
+
+    # The list of metrics to collect is appended to the pcp2json command
+    pcp2json_cmd.extend(metrics)
+
+    print(f"Reporting metrics for: {params.pmlogger_metrics}")
+
+    if params.flatten or params.generate_csv:
+        # Initialize (optional) pcp2csv command
+        pcp2csv_cmd = [
+            "/usr/bin/pcp2csv",
+            "-l",
+            ",",
+        ]
 
+        pcp2csv_cmd.extend(pcp2_flags)
+        pcp2csv_cmd.extend(metrics)
+
+    max_retries = 1
```

Comment:
I don't see any way to override the number of retries, so...unless you have future aspirations for this, now might be a good time to tear it out -- there's little point to having a _loop_ whose body is executed exactly **once**.  And, removing the retry logic will make this code considerably simpler.

Reply:
It was left flexible because I wasn't entirely sure that this solution was complete, but I'm confident enough with it at this point. However, this loop does run two iterations, not one, so this doesn't eliminate the retry logic.

Reply:
The loop expression,
```python
for _attempt in range(max_retries):
```
Runs _one_ iteration giving `_attempt` the value of `0` (because, after that, the next value would be `1` which is not in the range "from zero up to but not including `max_retries`", which is what the `range()` generator returns).  And, the `continue` statement in the loop proceeds with the _next_ iteration, of which there is none.

So, how does it run two iterations?

Reply:
I was under the impression (and I swore I even looked it up to be sure) that `range()` was inclusive of the upper bound, but it seems that is not the case and this has never worked as designed.